src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -1258,6 +1258,15 @@ public SeparatedSyntaxList<SyntaxNode> GetArgumentsOfObjectCreationExpression(Sy
         public SeparatedSyntaxList<SyntaxNode> GetArgumentsOfArgumentList(SyntaxNode? argumentList)
             => (argumentList as BaseArgumentListSyntax)?.Arguments ?? default;
 
+        public SyntaxNode GetArgumentListOfInvocationExpression(SyntaxNode invocationExpression)
+            => (invocationExpression as InvocationExpressionSyntax)!.ArgumentList;
+
+        public SyntaxNode GetArgumentListOfObjectCreationExpression(SyntaxNode objectCreationExpression)
+            => (objectCreationExpression as ObjectCreationExpressionSyntax)!.ArgumentList;
+
+        public SyntaxNode UpdateArgumentListSyntax(SyntaxNode argumentList, SeparatedSyntaxList<SyntaxNode> arguments)
+            => (argumentList as ArgumentListSyntax)!.WithArguments(arguments);
```

Comment:
remove.  move to subclasses of IntroduceParameterProvider

Reply:
done